test/unit/extract/eia_bulk_elec_test.py

```diff
@@ -0,0 +1,113 @@
+    {"series_id":"ELEC.COST_BTU.NG-US-2.A","name":"Average cost of fossil fuels for electricity generation (per Btu) : natural gas : United States : electric utility non-cogen : annual","units":"dollars per million Btu","f":"A","description":"Natural Gas; Power plants owned by regulated electric utilties that produce electricity only; ","copyright":"None","source":"EIA, U.S. Energy Information Administration","iso3166":"USA","geography":"USA","start":"2008","end":"2021","last_updated":"2022-02-25T15:25:17-05:00","geoset_id":"ELEC.COST_BTU.NG-2.A","data":[["2021",null],["2020",null]]}
+    {"series_id":"ELEC.RECEIPTS_BTU.NG-US-2.Q","name":"Receipts of fossil fuels by electricity plants (Btu) : natural gas : United States : electric utility non-cogen : quarterly","units":"billion Btu","f":"Q","description":"Natural Gas; Power plants owned by regulated electric utilties that produce electricity only; ","copyright":"None","source":"EIA, U.S. Energy Information Administration","iso3166":"USA","geography":"USA","start":"2008Q1","end":"2022Q1","last_updated":"2022-05-24T10:42:22-04:00","geoset_id":"ELEC.RECEIPTS_BTU.NG-2.Q","data":[["2021Q4",971009.16934],["2021Q3",1247751.19006],["2021Q2",924073.00164],["2021Q1",844665.11442],["2020Q4",972282.3858],["2020Q3",1439140.59549],["2020Q2",959842.7605],["2020Q1",1000417.64012]]}
+    {"series_id":"ELEC.RECEIPTS_BTU.NG-US-2.A","name":"Receipts of fossil fuels by electricity plants (Btu) : natural gas : United States : electric utility non-cogen : annual","units":"billion Btu","f":"A","description":"Natural Gas; Power plants owned by regulated electric utilties that produce electricity only; ","copyright":"None","source":"EIA, U.S. Energy Information Administration","iso3166":"USA","geography":"USA","start":"2008","end":"2021","last_updated":"2022-02-25T15:25:17-05:00","geoset_id":"ELEC.RECEIPTS_BTU.NG-2.A","data":[["2021",3987498.47545],["2020",4371683.38189]]}
+    """
+    return test_file
+
+
+@pytest.fixture()
+def elec_txt_dataframe(test_file_bytes) -> pd.DataFrame:
+    """Simulate raw pd.read_json('ELEC.txt')."""
+    buffer = BytesIO(test_file_bytes)
+    df = pd.read_json(buffer, lines=True)
+    return df
+
+
+def test__filter_df(elec_txt_dataframe):
+    """Filter for only the desired data series."""
+    input_ = elec_txt_dataframe
+    expected = input_.iloc[1:, :]
+    actual = bulk._filter_df(input_)
+    pd.testing.assert_frame_equal(actual, expected)
+
+
+def test__parse_data_column(elec_txt_dataframe):
+    """Convert a pd.Series of python lists to a dataframe."""
+    input_ = elec_txt_dataframe.iloc[[2, 4], :]  # only annual series for easier testing
```

Comment:
Here I'm subsetting the test fixture for easier testing. I'm picking out the two annual series because they are low resolution and thus less work to manually parse and write out the expected values. Maybe I should change my fixture setup so that this is easier to express?

Reply:
If there were some way to explicitly select the annual records based on their contents that would be clean. Or if there were a couple of variables / constants containing the different series (e.g. a monthly one and an annual one) taht could be addressed explicitly based on their frequency.